models/turbine_models/custom_models/stateless_llama.py

```diff
@@ -489,26 +491,362 @@ def evict_kvcache_space(self):
+            if not os.path.exists(self.external_weights_dir):
+                os.makedirs(external_weights_dir, exist_ok=True)
+            external_weight_path = os.path.join(
+                self.external_weights_dir, safe_name + self.external_weights
+            )
+        elif self.external_weights is None:
+            print(
+                "No external weights type specified using --external_weights, weights for imported .mlir files will not be externalized."
+            )
+            external_weight_path = None
+        else:
+            print(
+                f"No external weights directory specified using --external_weights_dir, we assume you have your own weights in {self.pipeline_dir}."
+            )
+            external_weights_dir = self.pipeline_dir
+            external_weight_path = os.path.join(
+                self.pipeline_dir, safe_name + self.external_weights
+            )
+        if weights_only:
+            input_mlir = None
+
+        _, vmfb = export_transformer_model(
+            self.hf_model_name,
+            hf_auth_token=self.hf_auth_token,
+            compile_to="vmfb",
+            external_weights=self.external_weights,
+            external_weight_file=external_weight_path,
+            quantization=quantization,
+            precision=self.precision,
+            device=self.device,
+            target_triple=self.iree_target_triple,
+            vulkan_max_allocation=None,
+            streaming_llm=True,
+            vmfb_path=os.path.join(self.pipeline_dir, safe_name + ".vmfb"),
+            upload_ir=False,
+            mod=None,
+            tokenizer=None,
+            decomp_attn=False,
+            input_mlir=input_mlir,
+            iree_flags=self.ireec_flags,
+        )
+        return vmfb, external_weight_path
+
+    # LOAD
+
+    def load_pipeline(
+        self,
+        vmfb: str,
+        weight: str,
+        rt_device: str = "local-task",
+        compiled_pipeline: bool = False,
+    ):
+        self.model = vmfbRunner(rt_device, vmfb, weight)
+
+    # RUN
+
+    def chat(self, prompt):
+        prompt = self.sanitize_prompt(prompt)
+
+        input_tensor = self.tokenizer(prompt, return_tensors="pt").input_ids
+
+        def format_out(results):
+            return torch.tensor(results.to_host()[0][0])
+
+        history = []
+        for iter in range(self.max_tokens):
+            # if self.streaming_llm:
+            token_slice = max(self.prev_token_len - 1, 0)
+            input_tensor = input_tensor[:, token_slice:]
+            # if self.streaming_llm and self.model["get_seq_step"]() > 600:
+            if self.model["get_seq_step"]() > 600:
+                print("Evicting cache space!")
+                self.model["evict_kvcache_space"]()
+            token_len = input_tensor.shape[-1]
+            device_inputs = [
+                ireert.asdevicearray(self.device, input_tensor)
+            ]
+            if self.first_input: # or not self.streaming_llm:
```

Comment:
streaming llm commented code? is this from the original?

Reply:
I redid some of the logic to remove non-streaming since I thought our plan was to only support streaming, but I think that's not actually the case. I'll add the support back in.